tests/service/XmlDocTests.fs

```diff
@@ -10,10 +10,119 @@ module Tests.Service.XmlDocTests.XmlDoc
+            SynModuleOrNamespace.SynModuleOrNamespace(decls = [
+                SynModuleDecl.Exception(range = range; exnDefn =
+                    SynExceptionDefn(range = exnDefnRange; exnRepr =
+                        SynExceptionDefnRepr(range = exnDefnReprRange)))])])) ->
+         Exception(range, exnDefnRange, exnDefnReprRange)
+
+     | ParsedInput.SigFile(ParsedSigFileInput(modules = [
+            SynModuleOrNamespaceSig.SynModuleOrNamespaceSig(decls = [
+                SynModuleSigDecl.Exception(range = range; exnSig =
+                    SynExceptionSig(range = exnSpfnRange; exnRepr =
+                        SynExceptionDefnRepr(range = exnDefnReprRange)))])])) ->
+         Exception(range, exnSpfnRange, exnDefnReprRange)
+
+     | x -> failwith $"Unexpected ParsedInput %A{x}"
+
+let (|UnionCases|) = function
+     | Types(_, [SynTypeDefn(typeRepr = SynTypeDefnRepr.Simple(simpleRepr = SynTypeDefnSimpleRepr.Union(_, cases, range)))])
+     | TypeSigs(_, [SynTypeDefnSig(typeRepr = SynTypeDefnSigRepr.Simple(repr = SynTypeDefnSimpleRepr.Union(_, cases, range)))]) ->
+         UnionCases(range, cases)
+
+     | x -> failwith $"Unexpected ParsedInput %A{x}"
+
+let (|Record|) = function
+    | Types(_, [SynTypeDefn(typeRepr = SynTypeDefnRepr.Simple(simpleRepr = SynTypeDefnSimpleRepr.Record(recordFields = fields)))])
+     | TypeSigs(_, [SynTypeDefnSig(typeRepr = SynTypeDefnSigRepr.Simple(repr = SynTypeDefnSimpleRepr.Record(recordFields = fields)))]) ->
+         Record(fields)
+
+     | x -> failwith $"Unexpected ParsedInput %A{x}"
+
+let (|Members|MemberSigs|) = function
+    | Types(_, [SynTypeDefn(typeRepr = SynTypeDefnRepr.ObjectModel(members = members))])
+    | Types(_, [SynTypeDefn(members = members)]) ->
+        Members(members)
+
+    | TypeSigs(_, [SynTypeDefnSig(typeRepr = SynTypeDefnSigRepr.ObjectModel(memberSigs = members))]) ->
+        MemberSigs(members)
+
+    | x -> failwith $"Unexpected ParsedInput %A{x}"
+
+let (|Lets|LetBindings|ValSig|LetOrUse|) = function
+    | ParsedInput.ImplFile(ParsedImplFileInput(modules = [
+            SynModuleOrNamespace.SynModuleOrNamespace(decls = [
+                SynModuleDecl.Let(bindings = [SynBinding(expr = SynExpr.LetOrUse(range = range; bindings = bindings))])])])) ->
+        LetOrUse(range, bindings)
+
+    | ParsedInput.ImplFile(ParsedImplFileInput(modules = [
+            SynModuleOrNamespace.SynModuleOrNamespace(decls = [
+                SynModuleDecl.Let(range = range; bindings = bindings)])])) ->
+        LetBindings(range, bindings)
+
+    | ParsedInput.ImplFile(ParsedImplFileInput(modules = [
+            SynModuleOrNamespace.SynModuleOrNamespace(decls = [
+                SynModuleDecl.DoExpr(expr = SynExpr.LetOrUse(range = range; bindings = bindings))])])) ->
+        LetBindings(range, bindings)
+
+    | ParsedInput.ImplFile(ParsedImplFileInput(modules = [
+            SynModuleOrNamespace.SynModuleOrNamespace(decls = decls)])) ->
+        Lets(decls)
```

Comment:
`Lets` doesn't sound like an suitable name here. Please, consider the following code which doesn't contain any `let` bindings:

```fsharp
module M =
    type T() = class end

    do ()

    open System
```